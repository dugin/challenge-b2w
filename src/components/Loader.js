import React from 'react';
import colors from "../styles/constants/colors";
import ReactLoading from 'react-loading';
import styled from 'styled-components';


const Container = styled.section`
 position:absolute;
 z-index: 1;
 width: 100%;
 height: 100%;
 display: flex;
 align-items: center;
 flex-direction: column;
 justify-content: center;
 
 & > *{
 margin-bottom: 10px;
 font-size: 14px;
 color:${colors.sapphire};
 font-weight: 300;
 }
`;



const Loader = (props) => {

    return (
        <Container>
            <ReactLoading type="spin" color={colors.sapphire} delay={0}  height={30} width={30}/>
            <span>Loading...</span>
        </Container>
    )
};
export default Loader;