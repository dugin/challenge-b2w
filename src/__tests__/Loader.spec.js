import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Loader from "../components/Loader";

describe('Loader Component', () => {

    const loader = shallow(<Loader/>);

    it('renders properly', () => {
        expect(toJson(loader)).toMatchSnapshot();
    });


});