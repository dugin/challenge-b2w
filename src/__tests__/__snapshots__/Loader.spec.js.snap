// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component renders properly 1`] = `
<styled.section>
  <Loading
    color="#00175B"
    delay={0}
    height={30}
    type="spin"
    width={30}
  />
  <span>
    Loading...
  </span>
</styled.section>
`;
